lib/metadataTypes/Filter.js

```diff
@@ -13,11 +39,176 @@ class Filter extends MetadataType {
      * but only with some of the fields. So it is needed to loop over
      * Filters with the endpoint /automation/v1/filters/{id}
      * @param {String} retrieveDir Directory where retrieved metadata directory will be saved
-     * @returns {Promise} Promise
+     * @returns {Promise<{metadata:FilterMap,type:string}>} Promise of items
      */
     static async retrieve(retrieveDir) {
         return super.retrieveREST(retrieveDir, '/automation/v1/filters/', null);
     }
+    /**
+     * manages post retrieve steps
+     * @param {FilterItem} item a single record
+     * @returns {FilterItem} parsed metadata definition
+     */
+    static postRetrieveTasks(item) {
+        return this.parseMetadata(item);
+    }
+    /**
+     * parses retrieved Metadata before saving
+     * @param {FilterItem} metadata a single record
+     * @returns {FilterItem} parsed metadata definition
+     */
+    static parseMetadata(metadata) {
+        try {
```

Comment:
this error is already handled in the saveResults method (only failing the individual record) so should be avoided here as will likely hide other errors/require additional checks in saveResults method